Tasks/SonarQubePreBuild/SonarQubeHelper.ps1

```diff
@@ -1,3 +1,20 @@
+#
+# Remarks: Some sensitive parameters cannot be stored on the agent between the 2 steps so 
+# we'll store them in the task context and pass them to the post-test step
+#
+function StoreSensitiveParametersInTaskContext
+{ 
+	param(
+		  [string]$serverUsername,
+		  [string]$serverPassword,
+		  [string]$dbUsername,
+		  [string]$dbPassword)
+
+	SetTaskContextVaraible "serverUsername" $serverUsername
+	SetTaskContextVaraible "serverPassword" $serverPassword
+	SetTaskContextVaraible "dbUsername" $dbUsername
+	SetTaskContextVaraible "dbPassword" $dbPassword
+}
 
 function CreateCommandLineArgs
```

Comment:
Pre-existing code: in what circumstances does it makes sense for the serverUrl to be null? It certainly doesn't make sense for hosted build agents, and it's questionable as to whether it should be allowed for hosted build agents (allowing users to rely on the http://localhost:9000 default doesn't seem to be a big win).